### ISSUE TITLE
Pressing Enter in translation field should insert newline ('\n')

### ISSUE THREAD

**Issue by mrjoops** [vslavik/poedit#144]
Would it be possible to handle new lines in translation ? It would be a great enhancement.

In v1.7.3, when I hit the "Enter" key on a translation field, it works as expected and I see the line break.
But when I switch to another translation and switch back to the previous one, the line break has disapeared. (No line break appears in the PO file).

This issue is also discussed on [stackoverflow](http://stackoverflow.com/questions/21682747/php-how-can-you-use-line-breaks-in-poedit)

**Comment by vslavik** [comment]
Poedit does, of course, handle multiline strings (as you are fully aware from SO); the markup to use is `\n` (ditto). Why don't you the describe the actual problem, instead of pretending to be unaware of the current solution and faking confusion?

The real issue being that, yes, Poedit should insert `\n` automatically when you hit Enter. Improvements in this area are planned for 1.8.

By the way: Please use other support channels for *questions*, only use the tracker to report actual bugs — i.e. *issues*. And when you do, please report things *well*, such as actually describing the issue (instead of linking to some long discussion where the problem appears as a minor footnote) and using a descriptive meaningful title that has relation to the problem

**Comment by mrjoops** [comment]
I actually described the problem, that's the point of the main sentence in my ticket.
And I don't pretend anything : SO provides a _workaround_, chich is quite far of a real solution, as the main users of poedit are non-developer people and inserting \n is out of question for them.
Anyway, I'm glad that an improvement is planned for 1.8.

**Comment by vslavik** [comment]
> I actually described the problem, that's the point of the main sentence in my ticket.

The subject was complete nonsense and I had no clue what you're talking about *at all* until I read that entire SO thread. So there's that.

> as the main users of poedit are non-developer people and inserting \n is out of question for them.

I love how you are so sure you know who the Poedit users are and what is or is not a problem for them. On one hand, you have a *feeling* that the current state is "out of question" for Poedit users. On the other hand, I have 15 years of hard *evidence* that it is not — apparently, non-developer translators are perfectly capable of realizing that `\n` in the source text needs to be translated that way in the translation. 

So: yes, this will be better in 1.8. But please don't blow it out of proportion. It really is not the end of the world you make it sound like.

**Comment by rulatir** [comment]
@vslavik The issue is described very unambiguously by the reporter, but let me rephrase it for you in bug report lingo:

Steps to reproduce:

1) Open a .po file that contains at least two translations
2) Start editing a translation by placing the caret in the translation field
3) Type a line of text
4) Press Enter
6) Type another line of text
7) Switch to another translation
8) Switch again to the translation you edited

Expected behavior:

The text should be formatted in two lines, just as you left it.

Actual behavior:

The line break is gone.

Rationale:

The perception that the actual behavior is wrong is based on several very general, common sense assumptions:

1) Software must obey. If I use software to introduce a change, the change must stick.

2) Visual representation should be meaningful. Different visual presentations of edited information should correspond to different content of the edited information. In particular, if a text field is showing me a visual line break at some point in the string, I should **not be wrong** to assume that what I see represents a different string value than a hypothetical one without a visual line break at that point. In combination with point 1), if I press Enter and the visual presentation of the edited data changes on screen (i.e. a line break appears), then I should be able to take it as a confirmation that my action has introduced meaningful, persistable change to the edited data.

3) An important use case of PoEdit is its use by translators. Translators could not care less about representation details such as `\n`. This should be completely hidden. A line break is a line break, and should be visually represented as an **actual** line break in a visual tool, full stop.

Remarks:

1) The original reporter is not "pretending to be unaware" of the current solution. The original reporter believes that the current solution is wrong, and I agree.

2) I tested the functionality in 2.0.5 and it is still unsatisfactory. The auto-inserted `\n` symbol, if shown at all, should behave as an atomic visual decoration. It should never be possible to put the editor widget in a state where a non-last line is not trailed by `\n`. It should never be possible to place the caret, or create a selection that begins or ends, in the middle of the `\n` symbol.

**Comment by vslavik** [comment]
> @vslavik The issue is described very unambiguously by the reporter, but let me rephrase

@rulatir Dear anonymous complainer, it may seem "unambiguous" to you, now three years later, but it sure as hell wasn't to me. And neither is your diatribe drive-by commenting on an outdated issue from time when *this behaved very differently from now*, despite you being so sure of yourself:

> Expected behavior:
> The text should be formatted in two lines, just as you left it.

And that indeed happens.

> Actual behavior:
> The line break is gone. 

This, though, does **NOT**. Or maybe it does, on some specific OS with some specific addons or what not, but you are so certain of my incompetency that you didn't bother providing even such basic information and didn't describe the system at all (let alone *well*). But it seems very unlikely. Anywhere I tested, Poedit behaves as expected. Which would be why this issue is, you know, *closed* fro three years.

> I tested the functionality in 2.0.5 and it is still unsatisfactory. The auto-inserted \n symbol, if shown at all, should behave as an atomic visual decoration

Thank you for educating me on the subject, oh The Wise One! If it weren't for your lecture, I would forever live in darkness about what the *ideal* implementation would be. …Or maybe, just maybe, some things are easier said than done due to technical limitations that you are wholly ignorant of. If you spent the time looking at the code instead of being jerk, perhaps you'd be able to offer a more valuable insight. But well, don't let the difficulty discourage you: get off your high horse and *do* something about what you think is poorly done; such PR would most definitely be very welcome.
